Evoscope branch

Maxime5G opened the pull request:
This PR adds evoscope (EVOlutionary Study of Correlations of Occurrences on a Phylogeny) to bioconda. Evoscope is a workflow to analyze correlated evolution between discrete traits reconstructed on a phylogenetic tree. 

Evo-Scope: Fully automated assessment of correlated evolution on phylogenetic trees
Maxime Godfroid, Charles Coluzzi, Amaury Lambert, Philippe Glaser, Eduardo P. C. Rocha, Guillaume Achaz
bioRxiv 2022.12.08.519595; doi: https://doi.org/10.1101/2022.12.08.519595 

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

Maxime5G commented:
@BiocondaBot please fetch artifacts

dpryan79 commented:
For whatever reason someone has to approve the CI jobs.

dpryan79 commented:
Ah, it's looking for gcc rather than respecting $CC

dpryan79 commented:
I see this has a shell script that runs some R scripts that are assumed to be in the current working directory. Practically speaking that will never be the case. I suggest putting the scripts directory in bin and then using something like the following to determine where that is:

```
SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
```

Maxime5G commented:
Thank you for looking at the recipe! I have now updated the main shell script to find the script directory using your suggested line - I also updated the build.sh file. Please let me know if that fixes the issue :)

dpryan79 commented:
Is there a good test for the R stuff that we could add?

dpryan79 commented:
Alternatively you could download and test one of the build artifacts.

Maxime5G commented:
I tried to add a test with R but somehow I couldn't manage to either execute the script (could not determine the path to the script for the test) nor find the test files. Anyway, I pushed a couple of changes to the build (new version), once that passes I'll test the build artefacts. Thanks again for the help!

Maxime5G commented:
@BiocondaBot please fetch artifacts
